pkg/parse_hal/hal_to_orca.go

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The Spinnaker Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package parse_hal
+
+import "github.com/spinnaker/kleat/api/client/config"
+
+func HalToOrca(h *config.Hal) *config.Orca {
+	return &config.Orca{
+		Default:           getDefaults(h),
+		PipelineTemplates: getPipelineTemplates(h),
+		Webhook:           h.GetWebhook(),
+	}
+}
+
+func getDefaults(h *config.Hal) *config.Orca_Defaults {
+	if !h.GetProviders().GetAws().GetEnabled() {
+		return nil
+	}
```

Comment:
I'm not sure if it's necessary to return `nil` if AWS is not enabled; I don't think there's any harm in setting this field anyway if AWS is not enabled from a functional perspective.  I did this mostly to avoid generating config that was not needed (similarly to how we set the pipeline template struct to `nil` if it's not set).  This really only affects humans reading the output of `kleat` so if we think we're adding complexity without equal benefit we can remove it and just always set the defaults to this return value below (and similar for pipeline templates).

Reply:
I like it the way it is; I think the complexity is worth it 🤷